### PR TITLE
Fix #3143 - InputText: focus on RTL still on left side

### LINKED ISSUE #3143: InputText: focus on RTL still on left side
## 1) Environment
- PrimeFaces version: 6.2-SNAPSHOT
- Does it work on the newest released PrimeFaces version? Version? No
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source) No
- Application server + version: Tomcat 7
- Affected browsers: Chrome

## 2) Expected behavior
I expect the focus to be on the right when `p:inputText` get the focus

## 3) Actual behavior
Focus still on the left

## 4) Steps to reproduce
Setting `primefaces.DIR` to rtl

## 5) Proposal
InputText should implement RTLAware and encode `dir` HTML attribute (see https://www.w3schools.com/cssref/pr_text_direction.asp)

### PR DESCRIPTION
**Pull request by Rapster**
I decided to put the method `encodeDirection` in `ComponentUtils` as it is a utility method. I could have write that in `CoreRenderer` too, but since most of the `CoreRenderer` component are not `RTLAware` I thought it gives more sense to put it in `ComponentUtils`. Let me know what do you think

### CODE REVIEW

File: src/main/java/org/primefaces/component/inputtext/InputTextRenderer.java [hunk 1]
```diff
@@ -73,6 +75,7 @@ protected void encodeMarkup(FacesContext context, InputText inputText) throws IO
             writer.writeAttribute("value", valueToRender, null);
         }
 
+        ComponentUtils.encodeDirection(context, inputText);
```
Comment (melloware) [inline src/main/java/org/primefaces/component/inputtext/InputTextRenderer.java]: @Rapster I see what you did here and your comment about it but I would still vote for renderRtlDirection() method to be consistent with others.  I know that putting that in the base class feels weird but not if we have more and more components using it?  I will wait for @tandraschko thoughts.

**Comment by tandraschko** [comment]
I would put that method in InputRenderer or something. ComponentUtils is a too big pool.

File: src/main/java/org/primefaces/component/inputtext/InputTextRenderer.java (hunk 1 above)
Comment (Rapster) [reply src/main/java/org/primefaces/component/inputtext/InputTextRenderer.java]: @melloware yes I will reuse same naming convention

**Comment by Rapster** [comment]
Yeah that's what I thought first, but `dir` attribute can be eligible to more than just inputs. But the truth is, if we start using this new attribute in existing RTL components I'm afraid it will mess the rendering. So I guess, I'll move the method in InputRenderer taking account @melloware comment

**Comment by tandraschko** [comment]
we can still move it to CoreRenderer instead of InputRenderer

**Comment by Rapster** [comment]
Let's keep it in InputRenderer for now, CoreRenderer is already a big tool too

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
Could you fix the conflicts?

**Comment by tandraschko** [comment]
@Rapster ?

**Comment by Rapster** [comment]
Yep sorry, I'll try to do that asap :wink:

**Comment by tandraschko** [comment]
Applied it

**Pull request closed without merging**